content/flux/v0.x/query-data/csv.md

```diff
@@ -0,0 +1,257 @@
+---
+title: Query CSV data sources
+list_title: CSV
+description: >
+  Use [`csv.from()`](/flux/v0.x/stdlib/csv/from/) and
+  [experimental `csv.from()`](/flux/v0.x/stdlib/experimental/csv/from/)
+  to query CSV data with Flux.
+  Query a CSV string, CSV file, or CSV data from a URL.
+menu:
+  flux_0_x:
+    name: CSV
+    parent: Query data sources
+weight: 103
+related:
+  - /flux/v0.x/stdlib/csv/from/
+  - /flux/v0.x/stdlib/experimental/csv/from/, Experimental csv.from() function
+  - /influxdb/cloud/reference/syntax/annotated-csv/
+list_code_example: |
+  ```js
+  import "csv"
+
+  csvData = "
+  #group,false,false,true,true,true,false,false
+  #datatype,string,long,string,string,string,long,double
+  #default,_result,,,,,,
+  ,result,table,dataset,metric,sensorID,timestamp,value
+  ,,0,air-sensors,humidity,TLM0100,1627049400000000000,34.79
+  ,,0,air-sensors,humidity,TLM0100,1627049700000000000,34.65
+  ,,1,air-sensors,humidity,TLM0200,1627049400000000000,35.64
+  ,,1,air-sensors,humidity,TLM0200,1627049700000000000,35.67
+  "
+  
+  csv.from(csv: csvData)
+  ```
+---
+
+Use [`csv.from()`](/flux/v0.x/stdlib/csv/from/) and
+[experimental `csv.from()`](/flux/v0.x/stdlib/experimental/csv/from/)
+to query CSV data with Flux.
+Query a CSV string, CSV file, or CSV data from a URL.
+Import the `csv` or `experimental/csv` package.
+
+- [CSV parsing modes](#csv-parsing-modes)
+- [Results structure](#results-structure)
+- [Examples](#examples)
+  - [Query an annotated CSV string](#query-an-annotated-csv-string)
+  - [Query a raw CSV string](#query-a-raw-csv-string)
+  - [Query CSV data from a file](#query-csv-data-from-a-file)
+  - [Query CSV data from a URL](#query-csv-data-from-a-url)
+
+## CSV parsing modes
+`csv.from()` supports two CSV parsing modes:
+
+- **annotations**: _(Default)_ Use CSV annotations to determine column data types and table grouping.
+- **raw**: Parse all columns as strings and use the first row as the **header row**
+  and all subsequent rows as data.
+
+{{% note %}}
+When using the **annotations** parsing mode, CSV data must include all annotation rows
+(`#datatype`, `#group`, and `#default`).
+{{% /note %}}
+
+## Results structure
+The structure of results returned by `csv.from()` depends on the
+[parsing mode](#csv-parsing-modes) used.
+
+- **annotations**: `csv.from()` returns a stream of tables grouped by columns
+  defined as `true` in the `#group` annotation row.
+- **raw**: `csv.from()`returns a stream of tables with no grouping (all rows are in a single table).
+  All data is formatted as strings.
+
+## Examples
```

Comment:
Maybe nice to say on top-level pages:
If you're just getting started, see how to run this Flux query in xyz....and link back to that doc